### PR TITLE
Ndev 420 reschedule neon transaction with the same resource

### PR DESCRIPTION
**Pull request by otselnik**

### CODE REVIEW

File: proxy/mempool/mempool_api.py [hunk 1]
```diff
@@ -47,9 +49,10 @@ class MPTxRequest(MPRequest):
     nonce: int = field(compare=True, default=None)
     signature: str = field(compare=False, default=None)
     neon_tx: NeonTx = field(compare=False, default=None)
-    neon_tx_exec_cfg: Optional[NeonTxExecCfg] = None
+    neon_tx_exec_cfg: NeonTxExecCfg = field(compare=False, default=None)
     sender_address: str = field(compare=False, default=None)
     gas_price: int = field(compare=False, default=None)
+    resource: OperatorResourceInfo = field(compare=False, default=None)
```
Comment (rozhkovdmitrii) [inline proxy/mempool/mempool_api.py]: Is it should be here? I'm currently do review and will suggest you the better way further

File: proxy/mempool/mempool_api.py [hunk 2]
```diff
@@ -2,10 +2,12 @@
 
 from dataclasses import dataclass, field
 from enum import IntEnum
-from typing import Any, Tuple, Optional
+from typing import Any, Tuple
 from abc import ABC, abstractmethod
 from asyncio import Task
 
+from proxy.mempool.operator_resource_mng import OperatorResourceInfo
```
Comment (rozhkovdmitrii) [inline proxy/mempool/mempool_api.py]: relative path

File: proxy/mempool/operator_resource_mng.py [hunk 3]
```diff
@@ -0,0 +1,215 @@
...
+
+    @property
+    def secret_key(self) -> bytes:
+        return self.signer.secret_key()
+
+    @staticmethod
+    def _get_current_time() -> int:
+        return math.ceil(datetime.now().timestamp())
+
+    def init_perm_accounts(self, solana: SolanaInteractor) -> bool:
```
Comment (rozhkovdmitrii) [inline proxy/mempool/operator_resource_mng.py]: Move it into OperatorResourceMng please, as well as all the stuff connected with managing. Leave it plain old data please to not confuse guys who will work on the mempool executor site ). Only the things those are needed there.

File: proxy/mempool/operator_resource_mng.py [hunk 4]
```diff
@@ -0,0 +1,215 @@
...
+from logged_groups import logged_group
+from solana.account import Account as SolanaAccount
+from solana.publickey import PublicKey
+
+from ..common_neon.address import EthereumAddress, ether2program
+from ..common_neon.compute_budget import TransactionWithComputeBudget
+from ..common_neon.constants import STORAGE_SIZE, ACTIVE_STORAGE_TAG, FINALIZED_STORAGE_TAG, EMPTY_STORAGE_TAG
+from ..common_neon.solana_tx_list_sender import SolTxListSender
+from ..common_neon.environment_utils import get_solana_accounts
+from ..common_neon.environment_data import EVM_LOADER_ID, PERM_ACCOUNT_LIMIT, RECHECK_RESOURCE_LIST_INTERVAL
```
Comment (rozhkovdmitrii) [inline proxy/mempool/operator_resource_mng.py]: In the mempool we use Config class to get things like this one

File: proxy/mempool/mempool_api.py (hunk 1 above)
Comment (otselnik) [reply proxy/mempool/mempool_api.py]: IMO - yes

File: proxy/mempool/mempool_api.py (hunk 2 above)
Comment (otselnik) [reply proxy/mempool/mempool_api.py]: fixed in [NDEV-419](https://github.com/neonlabsorg/proxy-model.py/pull/869) PR

File: proxy/mempool/operator_resource_mng.py (hunk 3 above)
Comment (otselnik) [reply proxy/mempool/operator_resource_mng.py]: Need to be discussed(in https://github.com/neonlabsorg/proxy-model.py/pull/869 PR)

File: proxy/mempool/mempool_schedule.py [hunk 5]
```diff
@@ -93,21 +93,25 @@ def drop_request_away(self, mp_tx_request: MPTxRequest):
...
         if self._processing_tx is None:
             self.error(f"Failed to reschedule tx with nonce: {nonce}, processing tx is None")
             return
         if self._processing_tx.nonce != nonce:
             self.error(f"Failed to reschedule tx, processing tx has different nonce: {self._processing_tx.nonce} than: {nonce}")
             return
-
-        self.debug(f"Reset processing tx back to pending: {self.sender_address} - {self._txs[0].log_str}")
         self._processing_tx = None
+        index = bisect.bisect_left(self._txs, mp_tx_request)
```
Comment (afalaleev) [inline proxy/mempool/mempool_schedule.py:110]: It is impossible to execute another transaction except the first in the queue. 
So there are no reasons to find the transaction in the queue, all previous checks are correct.
And this check is surplus.

**Review by afalaleev — CHANGES_REQUESTED** [review]

**Comment by otselnik** [comment]
Changes implemented in
https://github.com/neonlabsorg/proxy-model.py/pull/869

**Pull request closed without merging**